### PR TITLE
Add IObserver interface for listening to Observer neighbor updates

### PR DESCRIPTION
**Pull request by mezz**
PR from @asiekierka 

This allows modded blocks to receive the exact same updates that the new Observers do.

Tested and it works as expected.

### CODE REVIEW

**Comment by liach** [comment]
Is there anything wrong with the pull?

**Comment by williewillus** [comment]
If we're going to pull the method up manually shouldn't we give it a nice name?

**Comment by mezz** [comment]
I think we probably need to update Forge's MCP before I can do the SRG magic to pull the method name up and have it obfuscate correctly.
I may end up just adding a new method again, but that decision is up to Lex at this point.

File: patches/minecraft/net/minecraft/block/Block.java.patch [hunk 1]
```diff
@@ -1138,7 +1138,7 @@
 +     * @param neighborBlock The updated block.
 +     * @param neighborPos The updated block's position.
 +     */
-+    public void func_190962_b(IBlockState p_190962_1_, World p_190962_2_, BlockPos p_190962_3_, Block p_190962_4_, BlockPos p_190962_5_){}
++    public void onNeighborChangeObserver(IBlockState state, World world, BlockPos pos, Block neighborBlock, BlockPos neighborPos){}
```
Comment (liach) [inline patches/minecraft/net/minecraft/block/Block.java.patch]: I think naming this method `onNeighborChangeObserved` or `observeNeighborChange`.

Comment (mezz) [reply patches/minecraft/net/minecraft/block/Block.java.patch]: This is very specific to the Observer block, I want to make that as clear as possible while also making it clear that it is similar to `onNeighborChange`. I think grammar is not important here.

Comment (williewillus) [reply patches/minecraft/net/minecraft/block/Block.java.patch]: Regardless of grammar I don't think it's that clear on first sight. Is the observer the block that this is called on? I agree with `observedNeighborChange` or `onObservedChange`, anything that indicates who is the observer and who is being observed (rename the parameters appropriately too -> `observedPos`, `observedState`, `observerState`, etc.)

Comment (mezz) [reply patches/minecraft/net/minecraft/block/Block.java.patch]: Ok updated with better names.

**Pull request merged**